src/components/inputs/localized-money-input/localized-money-input.js

```diff
@@ -0,0 +1,307 @@
+            name={this.props.name}
+            value={this.props.value}
+            onChange={this.handleChange}
+            onBlur={this.props.onBlur}
+            isDisabled={this.props.isDisabled}
+            placeholder={this.props.placeholder}
+            hasError={this.props.hasError}
+            hasWarning={this.props.hasWarning}
+            {...filterDataAttributes(this.props)}
+          />
+        </div>
+        <div className={styles.commandsContainer}>
+          <div className={styles.commandsLeft}>
+            {do {
+              if (this.props.error) <div>{this.props.error}</div>;
+              else if (this.props.warning) <div>{this.props.warning}</div>;
+              else this.props.currenciesControl;
+            }}
+          </div>
+        </div>
+        {(this.props.error || this.props.warning) &&
+          this.props.currenciesControl}
+      </Spacings.Stack>
+    );
+  }
+}
+
+export class LocalizedMoneyInput extends React.Component {
+  static displayName = 'LocalizedMoneyInput';
+
+  static propTypes = {
+    id: PropTypes.string,
+    name: PropTypes.string,
+    // then input doesn't accept a "currencies" prop, instead all possible
+    // currencies have to exist (with empty or filled strings) on the value:
+    //   { EUR: 12.00, USD: 14.23, CAD: 13.70 }
+    value: PropTypes.objectOf(PropTypes.string).isRequired,
+    onChange: PropTypes.func,
+    selectedCurrency: PropTypes.string.isRequired,
+    onBlur: PropTypes.func,
+    onFocus: PropTypes.func,
+    hideExpansionControls: PropTypes.bool,
+    isDefaultExpanded: (props, propName, componentName, ...rest) => {
+      if (props.hideExpansionControls && typeof props[propName] === 'boolean') {
+        throw new Error(
+          oneLine`
+            ${componentName}: "${propName}" does not have any effect when
+            "hideExpansionControls" is set.
+          `
+        );
+      }
+      return PropTypes.bool(props, propName, componentName, ...rest);
+    },
+    isDisabled: PropTypes.bool,
+    placeholder: PropTypes.objectOf(PropTypes.string),
+    horizontalConstraint: PropTypes.oneOf(['xs', 's', 'm', 'l', 'xl', 'scale']),
+    hasError: PropTypes.bool,
+    hasWarning: PropTypes.bool,
+    errors: PropTypes.objectOf(PropTypes.node),
+    warnings: PropTypes.objectOf(PropTypes.node),
+    // HoC
+    intl: PropTypes.shape({
+      formatMessage: PropTypes.func.isRequired,
+    }).isRequired,
+  };
+
+  static getId = getId;
+
+  static getName = getName;
+
+  static defaultProps = {
+    horizontalConstraint: 'scale',
+  };
+
+  static convertToMoneyValues = values =>
+    Object.keys(values).map(currencyCode =>
+      MoneyInput.convertToMoneyValue({
+        currencyCode,
+        amount: values[currencyCode],
+      })
+    );
+
+  static parseMoneyValues = (moneyValues = [], locale) =>
+    moneyValues.map(value => MoneyInput.parseMoneyValue(value, locale));
+
+  static isHighPrecision = (values = []) =>
```

Comment:
As our `props.value` accepts an object, I'd make `isHighPrecision` work with an object as well.

So we should be able to do

```js
LocalizedMoneyInput.isHighPrecision({ EUR: "12.00", USD: "14.23323", CAD: "" })
// -> true
```

---

Another issue with the current implementation is that the name is `isHighPrecision` which indicates a boolean would be returned, but it returns an object.

---

Similar to how I described `LocalizedMoneyInput.getEmptyCurrencies` above, we can offer a way for our consumers to find out which of the supplied prices are high-precision. We can offer this API for it:

```js
LocalizedMoneyInput.getHighPrecisionCurrencies({ EUR: "12.00", USD: "14.2334", CAD: "" })
// -> ["USD"]
```

> Sidenote: The implementation of `LocalizedMoneyInput.isHighPrecision` can then just be
> ```js
> LocalizedMoneyInput.isHighPrecision = value =>
>   LocalizedMoneyInput.getHighPrecisionCurrencies(value).length > 0
> ```